pkg/primaza/controlplane/application_namespaces.go

```diff
@@ -132,11 +134,17 @@ func PushServiceCatalogToApplicationNamespaces(ctx context.Context, sc primazaio
 				Name:      sc.Name,
 				Namespace: ns,
 			},
-			Spec: sc.Spec,
 		}
 
-		if err := cli.Create(ctx, sccp, &client.CreateOptions{}); err != nil {
-			return err
+		op, err := controllerutil.CreateOrUpdate(ctx, cli, sccp, func() error {
+			sccp.Spec = sc.Spec
+			return nil
+		})
+
+		if err != nil {
+			l.Error(err, "Failed to create or update service catalog")
```

Comment:
should we return the error here?

```suggestion
			l.Error(err, "Failed to create or update service catalog")
			return err
```

Reply:
As we are creating/updating the catalog in a loop , let's create a error list and return that. WDYT?

Reply:
right, nice catch